lib/utils.js

```diff
@@ -13,6 +13,25 @@ module.exports = {
     });
   },
 
+	getApiGatewayStageName() {
+		let stageNameStage, stageNameDeploy;
+		const serviceTemplate = this.serverless.service.resources;
+		Object.keys(serviceTemplate.Resources).forEach(function(key){
+			if (serviceTemplate.Resources[key]['Type'] == 'AWS::ApiGateway::Stage') {
```

Comment:
Here, `key` would represent the logical ID that may or may not need to be added to DependsOn.